lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -64,6 +65,11 @@ EfficientGlobalOptimization::EfficientGlobalOptimization(const OptimizationProbl
 {
   checkProblem(problem);
   if (krigingResult_.getMetaModel().getOutputDimension() != 1) throw InvalidArgumentException(HERE) << "Metamodel must be 1-d";
+  if (noise.getEvaluation().getImplementation()->isActualImplementation())
+  {
+    setMetamodelNoise(noise);
+    setNoiseModel(noise);
+  }
```

Comment:
what is the difference between `noiseModel` and `metamodelNoise`?

Reply:
Citing Julien a few posts above:
> in case of AEI we wanted to provide 2 noise functions: one for costly evaluations and called only when a new point is added to the design, and one cheaper evaluation that can be called much more often for the optimization of the criterion in-between design increments

Reply:
ok I see the point. But as mentioned in the doc (see [openturns EGO](https://openturns.github.io/openturns/latest/user_manual/_generated/openturns.EfficientGlobalOptimization.html) for more details): 
```
A less computationally expensive noise function can be provided through [setNoiseModel()](https://openturns.github.io/openturns/latest/user_manual/_generated/openturns.EfficientGlobalOptimization.html#openturns.EfficientGlobalOptimization.setNoiseModel) to evaluate \sigma^2_{\epsilon}(x) for the improvement criterion optimization, the objective being only used to compute values and associated noise at design points.
```

So my comprehension is that user already provide a noise model and it is in a way his responsibility to make sure CPU time is compatible with what he is doing. If we start also to have some multi-fidelity here...

Reply:
My point is that `EGO` is already implementing two algorithms (`EI` and `AEI`). Now we are proposing 3 algorithms (`EI`, `AEI` and more optimal `AEI`). All in the same class

Reply:
Just to be clear, I didnt change the behavior of the class (there were already 2 levels of noise function), just the API to pass the metamodel noise.

Reply:
ok